File: staff_frontend/lib/models/cluster.dart
class ClusterStaffMember {
  final int employeeId;
  final String employeeName;
  final String? positionTitle;
  // True when here via an additional-store link (primary elsewhere) — NOT movable
  // from this store.
  final bool alsoCovers;

  const ClusterStaffMember({
    required this.employeeId,
    required this.employeeName,
    required this.positionTitle,
    required this.alsoCovers,
  });

  factory ClusterStaffMember.fromJson(Map<String, dynamic> j) =>
      ClusterStaffMember(
        employeeId: j['employee_id'] as int,
        employeeName: j['employee_name'] as String,
        positionTitle: j['position_title'] as String?,
        alsoCovers: (j['also_covers'] as bool?) ?? false,
      );
}

class ClusterStore {
  final int storeId;
  final String storeName;
  final int brandId;
  final String brandName;
  final List<ClusterStaffMember> staff;

  const ClusterStore({
    required this.storeId,
    required this.storeName,
    required this.brandId,
    required this.brandName,
    required this.staff,
  });

  factory ClusterStore.fromJson(Map<String, dynamic> j) => ClusterStore(
        storeId: j['store_id'] as int,
        storeName: j['store_name'] as String,
        brandId: j['brand_id'] as int,
        brandName: j['brand_name'] as String,
        staff: ((j['staff'] as List?) ?? const [])
            .map((e) => ClusterStaffMember.fromJson(e as Map<String, dynamic>))
            .toList(growable: false),
      );

  /// Staff whose primary store is here (the only ones movable from this store).
  List<ClusterStaffMember> get movable =>
      staff.where((s) => !s.alsoCovers).toList(growable: false);
}

File: backend/app/schemas/cluster.py
from pydantic import BaseModel


# ---- Cluster read view (Phase 2a) ----------------------------------------
class ClusterStaff(BaseModel):
    employee_id: int
    employee_name: str
    position_title: str | None = None
    # True when the staffer appears here via an additional-store link (their
    # primary store is elsewhere) — shown as an "also covers" tag, and NOT movable
    # from this store.
    also_covers: bool


class ClusterStore(BaseModel):
    store_id: int
    store_name: str
    brand_id: int
    brand_name: str
    staff: list[ClusterStaff]


class ClusterResponse(BaseModel):
    stores: list[ClusterStore]


# ---- Move staff (Flow 1) --------------------------------------------------
class MoveRequest(BaseModel):
    to_store_id: int


class MoveResult(BaseModel):
    employee_id: int
    employee_name: str
    from_store_id: int
    to_store_id: int
    to_store_name: str


# ---- Request staff (Flow 2) ----------------------------------------------
class StaffSearchResult(BaseModel):
    employee_id: int
    employee_name: str
    brand_names: list[str]
    store_names: list[str]


class RequestAssignmentRequest(BaseModel):
    store_id: int


class RequestAssignmentResult(BaseModel):
    status: str
    notification_id: int
